Makefile

```diff
@@ -97,6 +117,23 @@ else
 CONTROLLER_GEN=$(shell which controller-gen)
 endif
 
+# find or download golic 
+# download golic if necessary
+golic:
+ifeq (, $(shell which golic))
+	@{ \
+	set -e ;\
+	GOLIC_TMP_DIR=$$(mktemp -d) ;\
+	cd $$GOLIC_TMP_DIR ;\
+	go mod init tmp ;\
+	go get github.com/AbsaOSS/golic@v0.1.2 ;\
+	rm -rf $$GOLIC_TMP_DIR ;\
+	}
+GOLIC=$(GOBIN)/golic
+else
+GOLIC=$(shell which golic)
+endif
+
```

Comment:
We had a discussion with @k0da on this. Unfortunately, with  GO < 1.16 you can't get module without modifying `go.mod` file. @k0da's or kubebuilder code handles that ⬆️ .

With go 1.16 you can install module directly from github, and your go.mod stays unchanged:
```
go install github.com/AbsaOSS/golic@v0.1.2
golic inject -c="2021 ABSA Group Limited"
```